client/src/js/components/bhDepotDropdown.js

```diff
@@ -35,6 +36,13 @@ function bhDepotController(Depot, AppCache, Notify) {
   $ctrl.select = function select(option) {
     $ctrl.selection = option;
     cache.selection = $ctrl.selection;
-    $ctrl.onSelect({ depot: $ctrl.selection });
+    $ctrl.onSelect({ depot : $ctrl.selection });
   };
+
+  function cachedDepotExist() {
+    if (!cache.selection) { return false; }
+    return $ctrl.depots.map(function (item) {
```

Comment:
Ah, I misunderstood the code.  You are right ... but a Array.some() is even better!